test/infrastructure/docker/internal/webhooks/dockercluster_webhook.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhooks
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	infrav1 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1beta1"
+)
+
+// DockerCluster implements a validating and defaulting webhook for DockerCluster.
+type DockerCluster struct{}
+
+func (webhook *DockerCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&infrav1.DockerCluster{}).
+		WithDefaulter(webhook).
+		WithValidator(webhook).
+		Complete()
+}
+
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-infrav1-dockercluster,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=dockerclusters,versions=infrav1,name=default.dockercluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;infrav1
```

Comment:
Please add `		paths=./internal/webhooks/... \` to l.343 in the Makefile.

This would have surfaced that this change leads to changes in the manifests.yaml

Reply:
Please also change l.336 in the Makefile to
```
	$(MAKE) clean-generated-yaml SRC_DIRS="$(CAPD_DIR)/config/crd/bases,$(CAPD_DIR)/config/webhook/manifests.yaml"
```

This ensures that the generation actually has to produce the manifests file 

(Please do something similar for all other generate-manifests-* targets)

Reply:
which other generate-manifests* are applicable for docker infra API(couldn't find those)

Reply:
All generate-manifests also the ones not for CAPD. Basically all of them should have a webhook/manifests.yaml file

Reply:
Shall I add all those in this PR itself?

Reply:
Yup

Reply:
Or up to you. You can also open a separate PR for the others

Reply:
Its good for me to add here 👍

Reply:
Let's merge this PR as it's otherwise ready. Feel free to include in one of your others or a separate one :)

Reply:
What I meant is that we should have the `config/webhook/manifests.yaml` cleanup for every generate-manifests target:
```
$(MAKE) clean-generated-yaml SRC_DIRS="$(CAPD_DIR)/config/crd/bases,$(CAPD_DIR)/config/webhook/manifests.yaml"
```

Currently we only have it for CAPD, e.g.
```
generate-manifests-kubeadm-control-plane: $(CONTROLLER_GEN) ## Generate manifests e.g. CRD, RBAC etc. for kubeadm control plane
	$(MAKE) clean-generated-yaml SRC_DIRS="./controlplane/kubeadm/config/crd/bases"
```

But it's entirely fine to do this in a separate PR